relay-server/src/services/processor/span/processing.rs

```diff
@@ -414,12 +414,15 @@ fn get_normalize_span_config<'a>(
 fn set_segment_attributes(span: &mut Annotated<Span>) {
     let Some(span) = span.value_mut() else { return };
 
-    // TODO: A span might be a segment span even if the parent_id is not empty
-    // (parent within a trace). I.e. do not overwrite here.
-    let is_segment = span.parent_span_id.is_empty();
+    let is_definitely_segment = span.parent_span_id.is_empty() || span.segment_id == span.span_id;
```

Comment:
Hehe nice name

Reply:
Hmm the second condition is also true when both are `None`... I'll add some more test coverage to not mess this up.